src/legacy/server/kbn_server.d.ts

```diff
@@ -45,7 +45,8 @@ import { IndexPatternsServiceFactory } from './index_patterns';
 import { Capabilities } from '../../core/server';
 import { UiSettingsServiceFactoryOptions } from '../../legacy/ui/ui_settings/ui_settings_service_factory';
 
-export type KibanaConfig = LegacyConfig;
+// lot of legacy code was assuming this type only had these two methods
+export type KibanaConfig = Pick<LegacyConfig, 'get' | 'has'>;
```

Comment:
Added `set` method in `LegacyConfig` to set the uuid in it for legacy compatibility. However a LOT of code in legacy made the assumption that `KibanaConfig` only had `get` and `has`.

Reply:
I'm surprised having `set` would break anything?

Reply:
Lots of TS errors similar to:
```
Type '() => { get<T>(key: string): any; has(key: string): boolean; }' is not assignable to type '() => LegacyConfig'.
        Property 'set' is missing in type '{ get<T>(key: string): any; has(key: string): boolean; }' but required in type 'LegacyConfig'
```

As `set` isn't used in legacy anyway, It felt like is was more pragmatic to restrict the type than to adapt something like 25 usages in multiple solutions code.